File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/Message.m
//
//  Message.m
//  OpenVideoCall
//
//  Created by 3 on 2020/12/15.
//  Portions Copyright (c) 2020 wuji-co. All rights reserved.
//

#import "Message.h"

@implementation Message
+ (instancetype)messageWithText:(NSString *)text type:(MessageType)type {
    Message *message = [[Message alloc] init];
    message.text = text;
    message.type = type;
    return message;
}
@end

File: Group-Video/OpenVideoCall-macOS/OpenVideoCall/Message.swift
//
//  Message.swift
//  OpenVideoCall
//
//  Created by 3 on 2020/12/16.
//  Copyright © 2020年 Wuji. All rights reserved.
//

import Cocoa

enum MessageType {
    case info, alert
    
    func color() -> NSColor {
        switch self {
        case .info: return NSColor(hex: 0x444444, alpha: 0.6)
        case .alert: return NSColor(hex: 0xff3c32, alpha: 0.6)
        }
    }
}

struct Message {
    var text: String!
    var type: MessageType = .info
    
    init(text: String, type: MessageType) {
        self.text = text
        self.type = type
    }
}
    
